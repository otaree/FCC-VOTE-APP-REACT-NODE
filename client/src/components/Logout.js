import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

import * as authActions from '../store/actions/auth';

export class Logout extends React.Component {
   async componentDidMount () {
        try {
            await axios({ url: "http://localhost:5000/user/logout", method: 'delete', headers: { 'x-auth': this.props.token } });
            this.props.logout();
            this.props.history.push("/");
        } catch (e) {
            this.props.logoutFail();
        }
    }
    render() {
        return (
            <div>
                {
                    this.props.authError.isError ?
                    <div>{this.props.authError.value}</div> :
                    <div>Loading...</div>
                }
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        authError: state.error,
        token: state.token
    };
};

const mapDispatchToProps = dispatch => {
    return {
        logout: () => dispatch(authActions.logout()),
        logoutFail: () => dispatch(authActions.fail("Unable to logout."))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Logout);

