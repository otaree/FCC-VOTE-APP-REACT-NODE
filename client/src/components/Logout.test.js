import React from 'react';
import { shallow } from 'enzyme';

import { Logout } from './Logout';

describe("Logout", () => {
    const props = {
        error: {
            isError: false
        }
    }
    const logout = <Logout {...props} />;

    it('renders correctly', () => {
        expect(logout).toMatchSnapshot();
    });
});