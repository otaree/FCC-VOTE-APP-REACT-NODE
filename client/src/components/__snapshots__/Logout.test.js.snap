// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<Logout
  error={
    Object {
      "isError": false,
    }
  }
/>
`;
